### PR TITLE
added __enter__ and __exit__ methods

### PR DESCRIPTION
**Pull request by zamai**
Adding __enter__ and __exit__ methods to the WebDriver class to create support for the with function in python
 https://docs.python.org/3/reference/compound_stmts.html#with

- [x] By placing an `X` in the preceding checkbox, I verify that I have signed the [Contributor License Agreement](https://github.com/SeleniumHQ/selenium/blob/master/CONTRIBUTING.md#step-6-sign-the-cla)

### CODE REVIEW

**Comment by davehunt** [comment]
Why not add these to the base class, so all sub-classes inherit it? Also, could you add tests? I know this has been proposed before, and although I have no objection to it, I wonder if there's a reason not to add it that I'm unaware of. @AutomatedTester?

File: py/selenium/webdriver/phantomjs/webdriver.py [hunk 1]
```diff
@@ -61,6 +61,15 @@ def __init__(self, executable_path="phantomjs",
             raise
 
         self._is_remote = False
+    """ 
+    In order for the python with statement to work __enter__ and __exit__ methods must be deffined in the object.
```
Comment (cgoldberg) [inline py/selenium/webdriver/phantomjs/webdriver.py]: i don't think this comment is needed.  if anything, I would just state "__enter__ and __exit__ methods enable using this class as a context manager"

Comment (zamai) [inline py/selenium/webdriver/phantomjs/webdriver.py]: Removed comment.

**Comment by zamai** [comment]
@davehunt I've added the methods to the Class level.
But I'm not sure regarding the tests, how exactly should I test this? 
Thanks,

**Comment by davehunt** [comment]
> But I'm not sure regarding the tests, how exactly should I test this?

Add a test that using `with` to exercise this change. I'm also keen to hear from @AutomatedTester if he's comfortable with this change.

**Comment by AutomatedTester** [comment]
I dont mind this change.

**Comment by lmtierney** [comment]
@zamai are you still planning on writing tests for this?

**Comment by zamai** [comment]
@lmtierney hi, I'm planning to write them this weekend, will let you know here

**Comment by cgoldberg** [comment]
@zamai, small nitpicks:
The linter (flake8) is failing on your branch.  Could you fix the formatting?  You should add 2 blank lines before the `__enter__` method and remove the trailing whitespace from the code inside both methods.
thanks!

**Comment by zamai** [comment]
Hey, @cgoldberg and @lmtierney 
I've updated the PR. 
Now there are not flake8 errors, from what I see
Thanks for helping me with it

**Comment by cgoldberg** [comment]
lgtm

**Comment by lmtierney** [comment]
@zamai This still needs a simple test, please update

**Comment by lmtierney** [comment]
Closed in favor of #5919

**Pull request closed without merging**